### ISSUE TITLE
[Bug] Divine Shield (BB #712)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#712]
This issue was migrated from bitbucket.
**Original Reporter:** Flywithme
**Original Date:** 09.10.2014 17:16:54 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712

<hr>

http://tbc.wowroad.info/?spell=642 (Protects the paladin from **all damage and spells** for 10 sec) - Not working properly.

Damage from Blessing of Sacrifice (http://tbc.wowroad.info/?spell=6940) passes through Divine Shield.
![WoWScrnShot_100914_200447.jpg](https://bitbucket.org/repo/kMeMB/images/3738675599-WoWScrnShot_100914_200447.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.10.2014 18:02:28 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-12878585

<hr>

This is because handling of SPELL_AURA_SPLIT_DAMAGE_\* ignores immunities and absorbs

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 09.10.2014 18:28:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-12879063

<hr>

This is perhaps the only spell which has such spell aura.

I did not know that Divine Shield can skip though what that damage :)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.10.2014 19:11:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-12879747

<hr>

It is indeed an issue, but I'm just stating why it's an issue so someone can fix it

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 10.10.2014 17:18:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-12898702

<hr>

the same problem with the spell http://tbc.wowroad.info/?spell=31892 - reverse damage passes through Divine Shield.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 12.01.2015 10:12:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-14749472

<hr>

Unit.cpp

```
#!c++


             // Damage can be splitted only if aura has an alive caster
             Unit *caster = (*i)->GetCaster();
-            if (!caster || caster == pVictim || !caster->IsInWorld() || !caster->isAlive())
+            if (!caster || caster == pVictim || !caster->IsInWorld() || !caster->isAlive() || caster->IsImmunedToDamage((SpellSchoolMask)(*i)->GetSpellProto()->SchoolMask))
                 continue;

             int32 currentAbsorb;
@@ -2112,7 +2112,7 @@

             // Damage can only be split if the aura has an alive caster linked
             Unit *caster = (*i)->GetCaster();
-            if (!caster || caster == pVictim || !caster->IsInWorld() || !caster->isAlive())
+            if (!caster || caster == pVictim || !caster->IsInWorld() || !caster->isAlive() || caster->IsImmunedToDamage((SpellSchoolMask)(*i)->GetSpellProto()->SchoolMask))
                 continue;

             uint32 splitted = CalculatePctU(RemainingDamage, (*i)->GetModifier()->m_amount);
```

Seal of Blood reverse damage don't immune Divine Shield.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 12.01.2015 17:43:03 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-14761866

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 12.01.2015 19:31:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-14763951

<hr>

Please dont change status to resolved unless the change is pushed to the core

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 13.01.2015 16:10:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/712/#comment-14786242

<hr>

Fix #712 - thanks to Alex_Step

→ <<cset 44e85d557199>>
